pkg/distsql/request_builder.go

```diff
@@ -341,6 +341,7 @@ func (builder *RequestBuilder) SetFromSessionVars(dctx *distsqlctx.DistSQLContex
 	builder.Request.StoreBusyThreshold = dctx.LoadBasedReplicaReadThreshold
 	builder.Request.RunawayChecker = dctx.RunawayChecker
 	builder.Request.TiKVClientReadTimeout = dctx.TiKVClientReadTimeout
+	builder.Request.MaxExecutionTime = dctx.MaxExecutionTime
```

Comment:
IMO adding a timeout to the context should be the final step. If we design with the expectation that the entire execution path checks for `ctx.Done` messages, perhaps we should review and document in the gitHub issue which tidb paths lack `ctx.Done` checks and list related TODO sub-tasks.

/cc @bb7133